kingpin/actors/group.py

```diff
@@ -89,13 +97,26 @@ def _build_actions(self):
         passed into this actors 'init_context' are also passed into the
         actors that we're intantiating.
         """
-        if not self.option('contexts'):
+        contexts = self.option('contexts')
+        if not contexts:
             return self._build_action_group(self._init_context)
 
+        if type(contexts) == list:
+            context_data = self.option('contexts')
+        elif type(contexts) == dict:
+            context_string = open(contexts['file']).read()
+            context_string = kp_utils.populate_with_tokens(
+                string=context_string,
+                tokens=contexts['tokens'],
+                strict=True)
+            context_data = demjson.decode(context_string)
+
         actions = []
-        for context in self.option('contexts'):
+        for context in context_data:
             combined_context = dict(self._init_context.items() +
                                     context.items())
+            self.log.debug('Init context: %s' % self._init_context.items())
```

Comment:
Use different phrases here -- I know they match the variable names, but to the "caller" of this code they wouldn't be terribly useful.